controllers/hetznerbaremetalremediation_controller_test.go

```diff
@@ -0,0 +1,428 @@
+						infrav1.HostAnnotation: fmt.Sprintf("%s/%s", testNs.Name, hostName),
+					}
+					Expect(testEnv.Create(ctx, hetznerBaremetalMachine)).To(Succeed())
+					Expect(testEnv.Create(ctx, hetznerBareMetalRemediation)).To(Succeed())
+
+					Eventually(func() bool {
+						if err := testEnv.Get(ctx, capiMachineKey, capiMachine); err != nil {
+							return false
+						}
+
+						return isPresentAndFalseWithReason(capiMachineKey, capiMachine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason)
+					}, timeout).Should(BeTrue())
+				})
+			})
+
+			Context("HetznerBaremetalHost exist", func() {
+				BeforeEach(func() {
+					hostKey = client.ObjectKey{Name: hostName, Namespace: testNs.Name}
+
+					By("creating HetznerBareMetalHost")
+					host = helpers.BareMetalHost(
+						hostName,
+						testNs.Name,
+						helpers.WithRootDeviceHintRaid(),
+						helpers.WithHetznerClusterRef(hetznerCluster.Name),
+					)
+					Expect(testEnv.Create(ctx, host)).To(Succeed())
+
+					By("creating hetznerBaremetalMachine")
+					hetznerBaremetalMachine.Annotations = map[string]string{
+						infrav1.HostAnnotation: fmt.Sprintf("%s/%s", testNs.Name, hostName),
+					}
+					Expect(testEnv.Create(ctx, hetznerBaremetalMachine)).To(Succeed())
+
+					By("ensuring host is provisioned")
+					Eventually(func() bool {
+						if err := testEnv.Get(ctx, hostKey, host); err != nil {
+							return false
+						}
+
+						testEnv.GetLogger().Info("Provisioning state of host", "state", host.Spec.Status.ProvisioningState, "conditions", host.Spec.Status.Conditions)
+						return host.Spec.Status.ProvisioningState == infrav1.StateProvisioned
+					}, timeout).Should(BeTrue())
+				})
+
+				AfterEach(func() {
+					Expect(testEnv.Cleanup(ctx, host, hetznerBareMetalRemediation, hetznerBaremetalMachine)).To(Succeed())
+				})
+
+				It("should create hetznerBareMetalRemediation object successfully", func() {
+					Expect(testEnv.Create(ctx, hetznerBareMetalRemediation)).To(Succeed())
+
+					Eventually(func() error {
+						return testEnv.Get(ctx, hetznerBaremetalRemediationkey, hetznerBareMetalRemediation)
+					}, timeout).Should(BeNil())
+				})
+
+				It("checks that, under normal conditions, reboot annotation, retryCount and lastRemediated are set", func() {
+					By("creating hetznerBareMetalRemediation object")
+					Expect(testEnv.Create(ctx, hetznerBareMetalRemediation)).To(Succeed())
+
+					By("checking if host remediation occurred")
+					Eventually(func() bool {
+						if err := testEnv.Get(ctx, hetznerBaremetalRemediationkey, hetznerBareMetalRemediation); err != nil {
+							return false
+						}
+
+						if err := testEnv.Get(ctx, hostKey, host); err != nil {
+							return false
+						}
+
+						testEnv.GetLogger().Info("host annotations", "hostAnnotation", host.Annotations)
+						return hetznerBareMetalRemediation.Status.LastRemediated != nil && hetznerBareMetalRemediation.Status.RetryCount == 1 && host.Annotations != nil
```

Comment:
don't you want to look for a specific annotation if you make the test in that way?